src/base/utils/string.cpp

```diff
@@ -130,6 +130,20 @@ bool NaturalCompare::lessThan(const QString &left, const QString &right)
     return false;
 }
 
+bool Utils::String::naturalCompareCaseSensitive(const QString &left, const QString &right)
+{
+    // provide a single `NaturalCompare` instance for easy use
+    static NaturalCompare nCmp(true);  // this is thread-safe in C++11 (stated in spec 6.7.4)
+    return nCmp(left, right);
+}
+
+bool Utils::String::naturalCompareCaseInsensitive(const QString &left, const QString &right)
+{
+    // provide a single `NaturalCompare` instance for easy use
+    static NaturalCompare nCmp(false);  // this is thread-safe in C++11 (stated in spec 6.7.4)
```

Comment:
good point, the `static NaturalCompare nCmp(false);` is thread-safe as I stated in comments.

but `return nCmp(left, right);` is not, yet it can easily fix by changing parameters to `(const QString left, const QString right)`